[Bug] Dimensionless (phantom) fins cause NaN exception when simulated

Reported to openrocket-bugs@lists.sourceforge.net by pga2266670@aol.com.

Attached is the ork file. no way to attach the bug report.
I tried the other option but it was a royal PITA.
The problem appeared when I tried to do a flight simulation.
 
Sincerely
Paul Gagnon

I reran his .ork and confirmed issue under 22.02.
---------- Exception stack trace ----------
net.sf.openrocket.gui.main.SwingExceptionHandler$InternalException: An exception occurred during the simulation
at net.sf.openrocket.gui.main.SwingExceptionHandler.handleErrorCondition(SwingExceptionHandler.java:120)
at net.sf.openrocket.gui.simulation.SimulationRunDialog$InteractiveSimulationWorker.simulationInterrupted(SimulationRunDialog.java:445)
at net.sf.openrocket.gui.simulation.SimulationWorker.done(SimulationWorker.java:109)
at java.desktop/javax.swing.SwingWorker$5.run(SwingWorker.java:750)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:847)
at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:857)
at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:317)
at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:249)
at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
at java.desktop/java.awt.Component.show(Component.java:1716)
at java.desktop/java.awt.Component.setVisible(Component.java:1663)
at java.desktop/java.awt.Window.setVisible(Window.java:1031)
at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
at net.sf.openrocket.gui.main.SimulationPanel.runSimulation(SimulationPanel.java:361)
at net.sf.openrocket.gui.main.SimulationPanel.access$900(SimulationPanel.java:74)
at net.sf.openrocket.gui.main.SimulationPanel$RunSimulationAction.actionPerformed(SimulationPanel.java:573)
at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
at java.desktop/java.awt.Component.processMouseEvent(Component.java:6635)
at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
at java.desktop/java.awt.Component.processEvent(Component.java:6400)
at java.desktop/java.awt.Container.processEvent(Container.java:2263)
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5011)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4843)
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4843)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:178)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:119)
at net.sf.openrocket.document.Simulation.simulate(Simulation.java:406)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:829)

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 22.02
OpenRocket source: default
OpenRocket location: /home/joseph/OpenRocket-22.02/OpenRocket.jar
User-defined thrust curves location: /home/joseph/.openrocket/ThrustCurves
JOGL version: 2.4.0-rc-20210111
Current default locale: en_US
System properties:
awt.toolkit=sun.awt.X11.XToolkit
exe4j.moduleName=/home/joseph/OpenRocket-22.02/OpenRocket
file.encoding=UTF-8
file.separator=/
install4j.appDir=/home/joseph/OpenRocket-22.02/
install4j.exeDir=/home/joseph/OpenRocket-22.02/
install4j.jvmDir=/home/joseph/OpenRocket-22.02/jre
install4j.launcherId=59
install4j.swt=false
java.awt.graphicsenv=sun.awt.X11GraphicsEnvironment
java.awt.printerjob=sun.print.PSPrinterJob
java.class.path=/home/joseph/OpenRocket-22.02/.install4j/i4jruntime.jar:/home/joseph/OpenRocket-22.02/.install4j/launcher47cd7162.jar:/home/joseph/OpenRocket-22.02/OpenRocket.jar
java.class.version=55.0
java.home=/home/joseph/OpenRocket-22.02/jre
java.io.tmpdir=/tmp
java.library.path=/usr/local/lib:/usr/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
java.net.useSystemProxies=true
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=11.0.17+7-LTS
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=11
java.util.Arrays.useLegacyMergeSort=true
java.vendor=BellSoft
java.vendor.url=https://bell-sw.com/
java.vendor.url.bug=https://bell-sw.com/support
java.version=11.0.17
java.version.date=2022-10-18
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=11
java.vm.vendor=BellSoft
java.vm.version=11.0.17+7-LTS
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln17270325696132891649
line.separator=\u000a
os.arch=amd64
os.name=Linux
os.version=6.1.0-1-amd64
path.separator=:
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=net.sf.openrocket.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=/home/joseph/OpenRocket-22.02/jre/lib
sun.cpu.endian=little
sun.cpu.isalist=
sun.font.fontmanager=sun.awt.X11FontManager
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=install4j.net.sf.openrocket.startup.OpenRocket
sun.java.launcher=SUN_STANDARD
sun.jnu.encoding=UTF-8
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.os.patch.level=unknown
user.country=US
user.dir=/home/joseph/OpenRocket-22.02
user.home=/home/joseph
user.language=en
user.name=joseph
user.timezone=America/Denver
---------- Error log ----------
1 0.001 INFO (SwingStartup.java:67) Starting up OpenRocket version 22.02
2 0.004 INFO (SwingStartup.java:75) Checking for graphics head
3 0.006 INFO (SwingStartup.java:86) Moving startup to EDT
4 0.458 INFO (SwingStartup.java:178) Initializing the splash screen
5 0.546 INFO (SwingStartup.java:182) Registering exception handler
6 0.550 INFO (SwingStartup.java:188) Loading databases
7 1.534 INFO (TranslatorProvider.java:49) Using default locale en_US
8 1.546 INFO (TranslatorProvider.java:61) Set up translation for locale en_US, debug.currentFile=messages.properties
9 1.547 INFO (ComponentPresetDatabaseLoader.java:94) Loading component presets from datafiles/components
10 1.547 INFO (MotorDatabaseLoader.java:89) Starting reading serialized motor database
11 1.550 INFO (SwingStartup.java:250) Starting update check
14 1.553 INFO (SwingStartup.java:201) Setting best LAF
18 1.717 INFO (MotorDatabaseLoader.java:95) Ending reading serialized motor database, motorCount=1896
19 1.719 INFO (MotorDatabaseLoader.java:70) Starting reading user-defined motors
20 1.728 INFO (MotorDatabaseLoader.java:80) Ending reading user-defined motors, motorCount=1903
84 2.085 INFO (SwingStartup.java:218) Opening main application window
86 2.093 INFO (BasicFrame.java:1671) New action initiated
101 0.000 DEBUG (-) ===== 82 DEBUG lines removed =====
101 2.152 DEBUG (Icons.java:24) Starting to load icons
102 2.167 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file publicmissiles-legacy.orc contained 115 presets
103 2.168 DEBUG (ComponentPresetDatabaseLoader.java:118) loading from file: publicmissiles.orc
104 2.168 DEBUG (OpenRocketComponentLoader.java:26) Loading presets from file publicmissiles.orc
105 2.190 DEBUG (Icons.java:109) Icons loaded
106 2.192 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file publicmissiles.orc contained 139 presets
107 2.193 DEBUG (ComponentPresetDatabaseLoader.java:118) loading from file: quest.orc
108 2.193 DEBUG (OpenRocketComponentLoader.java:26) Loading presets from file quest.orc
109 2.202 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file quest.orc contained 71 presets
110 2.202 DEBUG (ComponentPresetDatabaseLoader.java:118) loading from file: semroc-legacy.orc
111 2.202 DEBUG (OpenRocketComponentLoader.java:26) Loading presets from file semroc-legacy.orc
114 2.268 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file semroc-legacy.orc contained 874 presets
115 2.269 DEBUG (ComponentPresetDatabaseLoader.java:118) loading from file: semroc.orc
116 2.269 DEBUG (OpenRocketComponentLoader.java:26) Loading presets from file semroc.orc
118 2.340 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file semroc.orc contained 1676 presets
119 2.344 DEBUG (ComponentPresetDatabaseLoader.java:118) loading from file: top_flight.orc
120 2.344 DEBUG (OpenRocketComponentLoader.java:26) Loading presets from file top_flight.orc
121 2.347 DEBUG (OpenRocketComponentLoader.java:35) ComponentPreset file top_flight.orc contained 47 presets
122 2.351 DEBUG (ComponentPresetDatabaseLoader.java:60) Time to load presets: 804ms 5309 loaded from 31 files
124 2.413 DEBUG (GitHubAPIUtil.java:76) Update checker response code: 200
136 2.768 DEBUG (RocketFigure3d.java:141) Initializing RocketFigure3D OpenGL Canvas
137 2.768 DEBUG (RocketFigure3d.java:143) Setting up GL capabilities...
139 2.794 DEBUG (GitHubAPIUtil.java:76) Update checker response code: 200
140 2.800 DEBUG (ReleaseInfo.java:43) Invalid release tag format for release: OpenRocket_technical_documentation-v13.05
141 2.800 DEBUG (ReleaseInfo.java:43) Invalid release tag format for release: Development_of_an_Open_Source_model_rocket_simulation-thesis-v20090520
142 2.801 INFO (UpdateInfoRetriever.java:145) Current build is latest version
173 3.213 DEBUG (JarUtil.java:40) Found jar file using codeSource
174 3.213 DEBUG (ExampleDesignFile.java:118) Current jar file is: /home/joseph/OpenRocket-22.02/OpenRocket.jar
182 3.304 DEBUG (BasicFrame.java:299) ... Setting Initial Selection: [Rocket, Sustainer]
183 3.304 DEBUG (BasicFrame.java:303) BasicFrame instantiation complete
184 3.312 DEBUG (SwingStartup.java:303) Welcome dialog ignored
185 3.312 INFO (SwingStartup.java:226) Checking update status
186 3.313 INFO (SwingStartup.java:94) Startup complete
197 24.592 INFO (BasicFrame.java:1258) Starting OpenFileWorker
198 24.592 INFO (SwingWorkerDialog.java:133) Running SwingWorker net.sf.openrocket.gui.util.OpenFileWorker@74d3a703
199 24.595 INFO (OpenRocketLoader.java:44) Loading .ork file
200 24.643 DEBUG (DatabaseMotorFinder.java:48) type Single-use, manufacturer Cesaroni Technology, designation 30795O25000-P, diameter NaN, length NaN, digest 698fb5ef7da9e8c6887a0bd1d9c8dac7, warnings WarningSet[]
201 24.673 INFO (OpenRocketLoader.java:102) Loading done
202 24.693 INFO (SwingWorkerDialog.java:151) Worker completed before opening dialog
203 24.693 DEBUG (BasicFrame.java:1323) Opening new frame with the document
204 24.694 DEBUG (BasicFrame.java:174) Instantiating new BasicFrame
205 24.694 DEBUG (BasicFrame.java:185) Constructing the BasicFrame UI
221 24.829 DEBUG (RocketFigure3d.java:141) Initializing RocketFigure3D OpenGL Canvas
222 24.829 DEBUG (RocketFigure3d.java:143) Setting up GL capabilities...
239 0.000 VBOSE (-) ===== 95 VBOSE lines removed =====
239 24.834 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
240 24.834 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
241 24.834 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
242 24.834 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.util.ComponentChangeAdapter@5bcdbddb, current number of listeners is 11
243 24.845 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.util.ComponentChangeAdapter@71f478fa, current number of listeners is 12
244 24.849 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.util.ComponentChangeAdapter@26e49d06, current number of listeners is 13
245 24.849 VBOSE (DoubleModel.java:953) DoubleModel[RocketFigure:Rotation] adding listener (total 1): net.sf.openrocket.gui.components.UnitSelector[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=javax.swing.plaf.synth.SynthBorder@52fd7784,flags=8388608,maximumSize=,minimumSize=,preferredSize=,defaultIcon=,disabledIcon=,horizontalAlignment=LEADING,horizontalTextPosition=TRAILING,iconTextGap=4,labelFor=,text=,verticalAlignment=CENTER,verticalTextPosition=CENTER]
246 24.849 VBOSE (DoubleModel.java:953) DoubleModel[RocketFigure:Rotation] adding listener (total 2): javax.swing.JSlider$ModelListener@62916152
247 24.850 VBOSE (DoubleModel.java:953) DoubleModel[RocketFigure:Rotation] adding listener (total 3): javax.swing.plaf.basic.BasicSliderUI$Handler@5be96bd3
248 24.850 VBOSE (DoubleModel.java:975) DoubleModel[RocketFigure:Rotation] removing listener (total 2): javax.swing.plaf.basic.BasicSliderUI$Handler@5be96bd3
249 24.850 VBOSE (DoubleModel.java:953) DoubleModel[RocketFigure:Rotation] adding listener (total 3): javax.swing.plaf.basic.BasicSliderUI$Handler@770cba5f
250 24.891 INFO (RocketPanel.java:965) using pre-existing simulation
251 24.893 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
252 24.893 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.util.ComponentChangeAdapter@608f13be, current number of listeners is 14
253 24.893 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.gui.scalefigure.RocketPanel$4@36c3b088, current number of listeners is 15
254 24.893 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
255 24.900 DEBUG (JarUtil.java:40) Found jar file using codeSource
256 24.900 DEBUG (ExampleDesignFile.java:118) Current jar file is: /home/joseph/OpenRocket-22.02/OpenRocket.jar
257 24.912 VBOSE (Rocket.java:480) Added ComponentChangeListener net.sf.openrocket.gui.main.BasicFrame$1@7bc996ad, current number of listeners is 16
258 24.943 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 1.00000 user scale, 3779.53 overall scale
259 24.943 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.0973656 user scale, 367.996 overall scale
260 24.944 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.0973656 user scale, 367.996 overall scale
261 24.944 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.0973656 user scale, 367.996 overall scale
262 24.945 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.0973656 user scale, 367.996 overall scale
263 24.945 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.0973656 user scale, 367.996 overall scale
264 24.945 DEBUG (BasicFrame.java:299) ... Setting Initial Selection: [Space, Sustainer]
265 24.945 DEBUG (BasicFrame.java:303) BasicFrame instantiation complete
266 24.950 INFO (BasicFrame.java:1184) Closing window because it is replaceable
267 24.950 DEBUG (BasicFrame.java:1608) Disposing window
268 25.046 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
269 25.047 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
270 25.047 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
271 25.050 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
272 25.051 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
273 25.051 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
274 25.091 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
275 25.092 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
276 25.092 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
277 25.197 DEBUG (Simulation.java:404) Simulation: calling simulator
278 25.204 INFO (BasicEventSimulationEngine.java:117) >>Starting simulation of branch: Sustainer
279 25.204 VBOSE (BasicEventSimulationEngine.java:303) HandleEvents: current branch = Sustainer
280 25.205 VBOSE (BasicEventSimulationEngine.java:305) Obtained event from queue: FlightEvent[type=LAUNCH,time=0.0,source=Space,data=null]
281 25.205 VBOSE (BasicEventSimulationEngine.java:306) Remaining EventQueue = []
282 25.205 INFO (BasicEventSimulationEngine.java:331) Queueing Ignition Event for: Body Tube/ed67278b / 30795O25000-P - Armed @: 0.0
283 25.205 VBOSE (BasicEventSimulationEngine.java:389) Handling event FlightEvent[type=LAUNCH,time=0.0,source=Space,data=null]
284 25.205 VBOSE (BasicEventSimulationEngine.java:305) Obtained event from queue: FlightEvent[type=IGNITION,time=0.0,source=Body Tube,data=30795O25000-P]
285 25.205 VBOSE (BasicEventSimulationEngine.java:306) Remaining EventQueue = []
286 25.205 VBOSE (BasicEventSimulationEngine.java:389) Handling event FlightEvent[type=IGNITION,time=0.0,source=Body Tube,data=30795O25000-P]
287 25.205 INFO (BasicEventSimulationEngine.java:406) Igniting motor: Body Tube/ed67278b / 30795O25000-P - Armed @0.0
288 25.205 VBOSE (BasicEventSimulationEngine.java:177) Taking simulation step at t=0.0 altitude 0.0
289 25.208 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
290 27.417 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
291 32.569 DEBUG (Simulation.java:404) Simulation: calling simulator
292 32.570 INFO (BasicEventSimulationEngine.java:117) >>Starting simulation of branch: Sustainer
293 32.570 VBOSE (BasicEventSimulationEngine.java:303) HandleEvents: current branch = Sustainer
294 32.570 VBOSE (BasicEventSimulationEngine.java:305) Obtained event from queue: FlightEvent[type=LAUNCH,time=0.0,source=Space,data=null]
295 32.570 VBOSE (BasicEventSimulationEngine.java:306) Remaining EventQueue = []
296 32.570 INFO (BasicEventSimulationEngine.java:331) Queueing Ignition Event for: Body Tube/ed67278b / 30795O25000-P - Armed @: 0.0
297 32.570 DEBUG (SimulationRunDialog.java:478) LAUNCH
298 32.570 VBOSE (BasicEventSimulationEngine.java:389) Handling event FlightEvent[type=LAUNCH,time=0.0,source=Space,data=null]
299 32.570 VBOSE (BasicEventSimulationEngine.java:305) Obtained event from queue: FlightEvent[type=IGNITION,time=0.0,source=Body Tube,data=30795O25000-P]
300 32.570 VBOSE (BasicEventSimulationEngine.java:306) Remaining EventQueue = []
301 32.570 VBOSE (BasicEventSimulationEngine.java:389) Handling event FlightEvent[type=IGNITION,time=0.0,source=Body Tube,data=30795O25000-P]
302 32.570 INFO (BasicEventSimulationEngine.java:406) Igniting motor: Body Tube/ed67278b / 30795O25000-P - Armed @0.0
303 32.570 VBOSE (BasicEventSimulationEngine.java:177) Taking simulation step at t=0.0 altitude 0.0
304 32.571 VBOSE (AbstractScaleFigure.java:193) ____ Updating RocketFigure to: 0.100610 user scale, 380.257 overall scale
305 32.576 DEBUG (SimulationRunDialog.java:241) Progressbar value 0
306 32.576 DEBUG (SimulationRunDialog.java:246) No simulation status data available, setting empty labels
307 32.618 DEBUG (SimulationRunDialog.java:365) simulationStage boost: t=0.0 est=1.3132741671378534
308 32.618 DEBUG (SimulationRunDialog.java:455) Setting progress to 0 (real 0)
309 32.618 DEBUG (SimulationRunDialog.java:241) Progressbar value 0
310 32.620 ERROR (SwingExceptionHandler.java:119) An exception occurred during the simulation
> net.sf.openrocket.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value, please report a bug, c=(NaN,NaN,NaN,w=NaN)
> at net.sf.openrocket.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:221)
> at net.sf.openrocket.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:296)
> at net.sf.openrocket.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:121)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:178)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:119)
> at net.sf.openrocket.document.Simulation.simulate(Simulation.java:406)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
> at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> at java.base/java.lang.Thread.run(Thread.java:829)
311 32.620 INFO (SwingExceptionHandler.java:142) Running in EDT, showing dialog
312 32.620 INFO (SwingExceptionHandler.java:201) Showing Exception dialog
313 35.706 USER (SwingExceptionHandler.java:220) User requested sending bug report
314 35.709 DEBUG (JarUtil.java:40) Found jar file using codeSource
---------- End of bug report ----------
[Space.ork.txt](https://github.com/openrocket/openrocket/files/10712279/Space.ork.txt)

--- a/core/resources/l10n/messages.properties
+++ b/core/resources/l10n/messages.properties
@@ -1882,6 +1882,7 @@ Warning.PODSET_FORWARD = In-line podset forward of parent airframe component
 Warning.PODSET_OVERLAP = In-line podset overlaps parent airframe component
 Warning.THICK_FIN = Thick fins may not simulate accurately.
 Warning.JAGGED_EDGED_FIN = Jagged-edged fin predictions may be inaccurate.
+Warning.ZERO_AREA_FIN = Fins with zero area will not affect aerodynamics
 Warning.LISTENERS_AFFECTED = Listeners modified the flight simulation
 Warning.RECOVERY_DEPLOYMENT_WHILE_BURNING = Recovery device opened while motor still burning.
 Warning.FILE_INVALID_PARAMETER = Invalid parameter encountered, ignoring.

--- a/core/src/net/sf/openrocket/aerodynamics/Warning.java
+++ b/core/src/net/sf/openrocket/aerodynamics/Warning.java
@@ -382,6 +382,10 @@ public abstract class Warning {
 	////Jagged-edged fin predictions may be inaccurate.
 	public static final Warning JAGGED_EDGED_FIN = new Other(trans.get("Warning.JAGGED_EDGED_FIN"));
 	
+	/** A <code>Warning</code> that the fins have a zero area. */
+	////Fins with no area will not affect aerodynamics
+	public static final Warning ZERO_AREA_FIN = new Other(trans.get("Warning.ZERO_AREA_FIN"));
+	
 	/** A <code>Warning</code> that simulation listeners have affected the simulation */
 	////Listeners modified the flight simulation
 	public static final Warning LISTENERS_AFFECTED = new Other(trans.get("Warning.LISTENERS_AFFECTED"));

--- a/core/src/net/sf/openrocket/aerodynamics/barrowman/FinSetCalc.java
+++ b/core/src/net/sf/openrocket/aerodynamics/barrowman/FinSetCalc.java
@@ -84,7 +84,9 @@ public class FinSetCalc extends RocketComponentCalc {
 	public void calculateNonaxialForces(FlightConditions conditions, Transformation transform,
 			AerodynamicForces forces, WarningSet warnings) {
 		
-		if (span < 0.001) {
+		warnings.addAll(geometryWarnings);
+		
+		if (finArea < MathUtil.EPSILON) {
 			forces.setCm(0);
 			forces.setCN(0);
 			forces.setCNa(0);
@@ -96,12 +98,6 @@ public class FinSetCalc extends RocketComponentCalc {
 			forces.setCyaw(0);
 			return;
 		}
-
-		if ((bodyRadius > 0) && (thickness > bodyRadius / 2)){
-			// Add warnings  (radius/2 == diameter/4)
-			warnings.add(Warning.THICK_FIN);
-		}
-		warnings.addAll(geometryWarnings);
 		
 		//////// Calculate CNa.  /////////
 		
@@ -241,17 +237,26 @@ public class FinSetCalc extends RocketComponentCalc {
 		
 		Coordinate[] points = component.getFinPoints();
 		
-		// Check for jagged edges
+		// Check geometry
 		geometryWarnings.clear();
 		boolean down = false;
 		for (int i = 1; i < points.length; i++) {
 			if ((points[i].y > points[i - 1].y + 0.001) && down) {
-				geometryWarnings.add(Warning.JAGGED_EDGED_FIN);
+				geometryWarnings.add(Warning.JAGGED_EDGED_FIN, component.toString());
 				break;
 			}
 			if (points[i].y < points[i - 1].y - 0.001) {
 				down = true;
 			}
+		}
+
+		if (finArea < MathUtil.EPSILON) {
+			geometryWarnings.add(Warning.ZERO_AREA_FIN, component.toString());
+		}
+
+		if ((bodyRadius > 0) && (thickness > bodyRadius / 2)){
+			// Add warnings  (radius/2 == diameter/4)
+			geometryWarnings.add(Warning.THICK_FIN, component.toString());
 		}
 		
 		// Calculate the chord lead and trail positions and length
@@ -365,7 +370,6 @@ public class FinSetCalc extends RocketComponentCalc {
 		macLead *= dy;
 		area *= dy;
 		rollSum *= dy;
-		
 		macLength /= area;
 		macSpan /= area;
 		macLead /= area;
@@ -614,6 +618,11 @@ public class FinSetCalc extends RocketComponentCalc {
 
 	@Override
 	public double calculateFrictionCD(FlightConditions conditions, double componentCf, WarningSet warnings) {
+		// a fin with 0 area contributes no drag
+		if (finArea < MathUtil.EPSILON) {
+			return 0.0;
+		}
+		
 		double cd = componentCf * (1 + 2 * thickness / macLength) * 2 * finArea / conditions.getRefArea();
 		return cd;
 	}
@@ -622,6 +631,11 @@ public class FinSetCalc extends RocketComponentCalc {
 	public double calculatePressureCD(FlightConditions conditions,
 									  double stagnationCD, double baseCD, WarningSet warnings) {
 		
+		// a fin with 0 area contributes no drag
+		if (finArea < MathUtil.EPSILON) {
+			return 0.0;
+		}
+
 		double mach = conditions.getMach();
 		double cd = 0;
 		

